src/mathMLTree.js

```diff
@@ -143,7 +134,83 @@ class TextNode {
     }
 }
 
+/**
+ * This node represents a space, but may render as <mspace.../> or as text,
+ * depending on the width.
+ */
+class SpaceNode {
+    width: number;
+    character: ?string;
+
+    /**
+     * Create a Space node with width given in CSS ems.
+     */
+    constructor(width: number) {
+        this.width = width;
+        // See https://www.w3.org/TR/2000/WD-MathML2-20000328/chapter6.html
+        // for a table of space-like characters.  We consistently use the
+        // &LongNames; because Unicode does not have single characters for
+        // &ThickSpace; (\u2005\u200a) and all negative spaces.
+        if (width >= 0.05555 && width <= 0.05556) {
+            this.character = "&VeryThinSpace;";  // \u200a
+        } else if (width >= 0.1666 && width <= 0.1667) {
+            this.character = "&ThinSpace;";      // \u2009
+        } else if (width >= 0.2222 && width <= 0.2223) {
+            this.character = "&MediumSpace;";    // \u2005
+        } else if (width >= 0.2777 && width <= 0.2778) {
+            this.character = "&ThickSpace;";     // \u2005\u200a
+        } else if (width >= -0.05556 && width <= -0.05555) {
+            this.character = "&NegativeVeryThinSpace;";
+        } else if (width >= -0.1667 && width <= -0.1666) {
+            this.character = "&NegativeThinSpace;";
+        } else if (width >= -0.2223 && width <= -0.2222) {
+            this.character = "&NegativeMediumSpace;";
+        } else if (width >= -0.2778 && width <= -0.2777) {
+            this.character = "&NegativeThickSpace;";
+        } else {
+            this.character = null;
+        }
+    }
+
+    /**
+     * Converts the math node into a MathML-namespaced DOM element.
+     */
+    toNode(): Node {
+        if (this.character) {
+            return document.createTextNode(this.character);
+        } else {
+            const node = document.createElementNS(
+                "http://www.w3.org/1998/Math/MathML", "mspace");
+            node.setAttribute("width", this.width + "em");
+            return node;
+        }
+    }
+
+    /**
+     * Converts the math node into an HTML markup string.
+     */
+    toMarkup(): string {
+        if (this.character) {
+            return this.character;
```

Comment:
I'm a little confused about the correct markup here. Perhaps it should be \``<mtext>${this.character}</mtext>`\`, and then we need to remove this `<mtext>` wrapper if it gets included in an `<mo>` or other token element?

Reply:
I'm not sure how to do this either.  In the past I've look at what MathJax outputs, but `<mspace width="thinmathspace" />` doesn't seem ideal.  Is it bad if we include `<mtext>` nodes inside `<mo>` or other element types?